tonilopezmr/CaptchaPartTwo.kt

```diff
@@ -0,0 +1,58 @@
+package day1
+
+import org.junit.Assert.*
+import org.junit.Test
+
+
+class CaptchaPartTwo {
+
+    fun sequence(sequence: String) : Int {
+        if (sequence.length % 2 != 0) return 0
+
+        val displacement = sequence.length / 2
+        val captcha = sequence + sequence.substring(0, displacement)
+
+        return captcha.foldIndexed(0) { idx, acc, a ->
+            if (idx < sequence.length && a.toString().toInt() == captcha[idx + displacement].toString().toInt()) {
+                acc + a.toString().toInt()
+            } else {
+                acc
+            }
+        }
+    }
+
+    @Test
+    fun `the sequence number must be even`() {
+        assertEquals(0, sequence("1"))
+    }
+
+    @Test
+    fun `1212 produces 6`() {
+        assertEquals(6, sequence("1212"))
+    }
+
+    @Test
+    fun `1221 produces 0`() {
+        assertEquals(0, sequence("1221"))
+    }
+
+    @Test
+    fun `123425 produces 4`() {
+        assertEquals(4, sequence("123425"))
+    }
+
+    @Test
+    fun `123123 produces 12`() {
+        assertEquals(12, sequence("123123"))
+    }
+
+    @Test
+    fun `12131415 produces 4`() {
+        assertEquals(4, sequence("12131415"))
+    }
+
+    @Test
+    fun `the result of the day 1 part 2 is`() {
```

Comment:
Nice test 👍

Reply:
First time I run this test I expect 0 in the result when test fails with the real result I copy to expect number hahaha

Reply:
![](https://media.giphy.com/media/QbumCX9HFFDQA/giphy-downsized.gif)